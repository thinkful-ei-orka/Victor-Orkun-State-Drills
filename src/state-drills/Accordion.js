import React from 'react';

export default class Accordion extends React.Component {
     static defaultProps = {
          sections: []
     };

     state = {
          activeIndex: null
     }

     // event handler for making a section active
     sectionActiveHandler = (sectionIndex) => {
          this.setState({ activeIndex: sectionIndex })
     }

     //generate a section to be rendered in the list later
     //<li>, <button>
     generateSection(section, activeIndex, index) {
          return (
               <li className='Section' key={index}>
                    <button onClick={() => this.sectionActiveHandler(index)}>
                       {section.title}  
                    </button>
                    <p>{section.content}</p>
               </li>
          )
     }

// render our <ul> full of section elements
     render() {
          const { activeIndex } = this.state
          const { sections } = this.props
          return (
               <ul className='Accordion'>
                    {sections.map((section, index) =>
                    this.generateSection(section, activeIndex, index) 
                    )}
               </ul>
          )
     }
}