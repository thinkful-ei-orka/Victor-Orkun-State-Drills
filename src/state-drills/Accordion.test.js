import React from 'react';
import Accordion from './Accordion';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

const exampleSections = [
     {
       title: 'Section 1',
       content: 'Lorem ipsum dolor sit amet consectetur adipisicing elit.',
     },
     {
       title: 'Section 2',
       content: 'Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!',
     },
     {
       title: 'Section 3',
       content: 'Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?',
     },
   ]

   describe(`Accordion`, () => {
     it('renders empty when there is no prop', () => {
     const wrapper = shallow(<Accordion/>)
     expect(toJson(wrapper)).toMatchSnapshot() 
     })

     it('renders no sections', () => {
     const wrapper = shallow(<Accordion sections={exampleSections}/>)
     expect(toJson(wrapper)).toMatchSnapshot()
     })

     it('opens a clicked section', () => {
     const wrapper = shallow(<Accordion sections={exampleSections}/>)
     wrapper.find('button').at(1).simulate('click')
     expect(toJson(wrapper)).toMatchSnapshot()
     })

     it('only opens the last section', () => {
     const wrapper = shallow(<Accordion sections={exampleSections}/>)
     wrapper.find('button').at(1).simulate('click')
     wrapper.find('button').at(2).simulate('click')
     expect(toJson(wrapper)).toMatchSnapshot()
     })
   })