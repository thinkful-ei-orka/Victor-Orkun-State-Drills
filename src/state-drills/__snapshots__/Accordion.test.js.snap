// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion only opens the last section 1`] = `
<ul
  className="Accordion"
>
  <li
    className="Section"
    key="0"
  >
    <button
      onClick={[Function]}
    >
      Section 1
    </button>
    <p>
      Lorem ipsum dolor sit amet consectetur adipisicing elit.
    </p>
  </li>
  <li
    className="Section"
    key="1"
  >
    <button
      onClick={[Function]}
    >
      Section 2
    </button>
    <p>
      Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!
    </p>
  </li>
  <li
    className="Section"
    key="2"
  >
    <button
      onClick={[Function]}
    >
      Section 3
    </button>
    <p>
      Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?
    </p>
  </li>
</ul>
`;

exports[`Accordion opens a clicked section 1`] = `
<ul
  className="Accordion"
>
  <li
    className="Section"
    key="0"
  >
    <button
      onClick={[Function]}
    >
      Section 1
    </button>
    <p>
      Lorem ipsum dolor sit amet consectetur adipisicing elit.
    </p>
  </li>
  <li
    className="Section"
    key="1"
  >
    <button
      onClick={[Function]}
    >
      Section 2
    </button>
    <p>
      Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!
    </p>
  </li>
  <li
    className="Section"
    key="2"
  >
    <button
      onClick={[Function]}
    >
      Section 3
    </button>
    <p>
      Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?
    </p>
  </li>
</ul>
`;

exports[`Accordion renders empty when there is no prop 1`] = `
<ul
  className="Accordion"
/>
`;

exports[`Accordion renders no sections 1`] = `
<ul
  className="Accordion"
>
  <li
    className="Section"
    key="0"
  >
    <button
      onClick={[Function]}
    >
      Section 1
    </button>
    <p>
      Lorem ipsum dolor sit amet consectetur adipisicing elit.
    </p>
  </li>
  <li
    className="Section"
    key="1"
  >
    <button
      onClick={[Function]}
    >
      Section 2
    </button>
    <p>
      Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!
    </p>
  </li>
  <li
    className="Section"
    key="2"
  >
    <button
      onClick={[Function]}
    >
      Section 3
    </button>
    <p>
      Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?
    </p>
  </li>
</ul>
`;
